endaq/calc/stats.py

```diff
@@ -6,6 +6,342 @@
+        - 9806.65 to convert from g to millimeters (mm)
+    :param include_integration: include the calculations of velocity and displacement, default behavior
+    :param include_pseudo_velocity: include the more time consuming calculation of pseudo velocity, default behavior
+    :param damp: the damping coefficient `ζ`, related to the Q-factor by
+        `ζ = 1/(2Q)`; defaults to 0
+    :param init_freq: the initial frequency in the sequence; if `None`,
+        use the frequency corresponding to the data's duration, default is 1.0 Hz
+    :param bins_per_octave: the number of frequencies per octave
+    :param include_resultant: add a resultant (root sum of the squares) for each metric, calculated from the other input dataframe columns
+    :param display_plots: display plotly figures of the min/max envelope of acceleration, velocity, displacement and PVSS (default as False)
+    :return: a dataframe containing all the metrics, one computed per column of the input dataframe
+    """
+
+    #Remove Offset
+    if zero is not None:
+        if zero == "start":
+            df = df - df.iloc[0]
+        elif zero in ("mean", "median"):
+            zero = dict(mean=np.mean, median=np.median)[zero]
+            df = df - df.apply(zero, axis="index", raw=True)
+
+    #Apply Filters & Window
+    df = filters.butterworth(df, low_cutoff=highpass_cutoff, tukey_percent=tukey_percent)
+
+    #Integrate
+    data_list = [df]
+    if include_integration:
+        [accel, vel, disp] = integrate.integrals(
+            df, n=2, tukey_percent=tukey_percent, highpass_cutoff=highpass_cutoff)
+        data_list = [df, vel, disp]
+
+    #Calculate Peak & RMS
+    metrics = pd.DataFrame()
+    rms_start = int(tukey_percent/2*df.shape[0])
+    rms_end = df.shape[0] - rms_start
+    for (label,data,scale) in zip(
+            ['Acceleration','Velocity','Displacement'],
+            data_list,
+            [1,vel_disp_multiplier,vel_disp_multiplier]):
+
+        #Apply Scale
+        data = data * scale
+
+        #Display Plots
+        if display_plots:
+            rolling_min_max_envelope(data).update_layout(yaxis_title_text = label).show()
+
+        #Calculate Absolute Peak
+        peak = pd.DataFrame(data.abs().max()).reset_index()
+        peak.columns = ['variable','value']
+        if include_resultant:
+            peak = pd.concat([peak, pd.DataFrame({
+                'variable':['Resultant'],
+                'value':[np.sum(peak.value**2)**0.5]
+            })])
+        peak['calculation'] = 'Peak Absolute ' + label
+
+        #Calculate RMS
+        rms = pd.DataFrame(data.iloc[rms_start:rms_end].pow(2).mean()**0.5).reset_index()
+        rms.columns = ['variable','value']
+        if include_resultant:
+            rms = pd.concat([rms, pd.DataFrame({
+                'variable':['Resultant'],
+                'value':[np.sum(rms.value**2)**0.5]
+            })])
+        rms['calculation'] = 'RMS ' + label
```

Comment:
Generally, rather than using string concatenation to join strings, format strings or f-strings are prefered

```suggestion
        rms['calculation'] = f'RMS {label}'
        rms['calculation'] = 'RMS {}'.format(label)
```